src/tools/clippy/clippy_lints/src/returns.rs

```diff
@@ -196,15 +195,10 @@ fn check_final_expr<'tcx>(
         ExprKind::Block(block, _) => {
             check_block_return(cx, block);
         },
-        ExprKind::If(_, then, else_clause_opt) => {
+        ExprKind::If(_, then, _) => {
             if let ExprKind::Block(ifblock, _) = then.kind {
                 check_block_return(cx, ifblock);
             }
-            if let Some(else_clause) = else_clause_opt {
```

Comment:
Weird that removing this doesn't break any tests, but as long as it works, I'm happy with less code :+1:

Reply:
Yes this section is removed because there is no more `DesugaringKind::LetElse`.

However, this lint is applicable to a let...else followed by return. I can add this case to this lint later.

Reply:
Sorry, I shall retract that. This is actually checking something. This is now restored.